apps/api/src/modules/job/job.service.ts

```diff
@@ -1,15 +1,78 @@
 import { Injectable } from '@nestjs/common'
 
-import { EditJobDto } from './job.dto'
+import { EditJobDto, GetJobCardWithMaxPageDto, SearchJobDto } from './job.dto'
+import { JobRepository } from './job.repository'
 
 @Injectable()
 export class JobService {
+  constructor(private repository: JobRepository) {}
+
   create(createJobDto: EditJobDto) {
     return 'This action adds a new job'
   }
 
-  findAll() {
-    return `This action returns all job`
+  convertRequestToParams(searchJobDto: SearchJobDto) {
+    const params = {}
+    if (searchJobDto.limit) {
+      params['take'] = Number(searchJobDto.limit)
+    }
+    if (searchJobDto.page) {
+      params['skip'] = (searchJobDto.page - 1) * searchJobDto.limit
+    }
```

Comment:
If limit and page is default maybe doing this
```suggestion
   const params = {
      take: Number(searchJobDto.limit),
      skip: (searchJobDto.page - 1) * searchJobDto.limit,
    }
```